environment.yaml

```diff
@@ -3,11 +3,17 @@ channels:
   - defaults
 dependencies:
   - python>=3.10
+  - pip
   - numpy
   - scipy=1.10.1
-  - pytorch::pytorch==2.0.0
-  - pytorch::torchvision==0.15.0
-  - pytorch::torchaudio==2.0.0
-  - dglteam::dgl
+  - pytorch::pytorch==2.1.0
+  - pytorch::torchvision==0.16.0
```

Comment:
Install works. Running test fails

```
Traceback (most recent call last):
  File "/Users/eslt0101/Projects/REPOS/eScience/SDCCA/test_DGL-PTM/DGL-PTM/dgl_ptm/dgl_ptm/model/initialize_model.py", line 324, in step
    ptm_step(self.model_graph,self.model_data,self.step_count,self.steering_parameters)
  File "/Users/eslt0101/Projects/REPOS/eScience/SDCCA/test_DGL-PTM/DGL-PTM/dgl_ptm/dgl_ptm/model/step.py", line 34, in ptm_step
    link_deletion(agent_graph, deletion_prob = params['deletion_prob'])
  File "/Users/eslt0101/Projects/REPOS/eScience/SDCCA/test_DGL-PTM/DGL-PTM/dgl_ptm/dgl_ptm/network/link_deletion.py", line 16, in link_deletion
    agent_graph.remove_edges(_edgeids_2_delete(agent_graph, deletion_prob))
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/eslt0101/Projects/REPOS/eScience/SDCCA/test_DGL-PTM/DGL-PTM/dgl_ptm/dgl_ptm/network/link_deletion.py", line 29, in _edgeids_2_delete
    upper_triangular = _sparse_upper_triangular(agent_graph.adj())
                                                ^^^^^^^^^^^^^^^^^
  File "/Users/eslt0101/mambaforge/envs/dgl_ptm_test/lib/python3.11/site-packages/dgl/heterograph.py", line 3821, in adj
    from .sparse import spmatrix
  File "/Users/eslt0101/mambaforge/envs/dgl_ptm_test/lib/python3.11/site-packages/dgl/sparse/__init__.py", line 43, in <module>
    load_dgl_sparse()
  File "/Users/eslt0101/mambaforge/envs/dgl_ptm_test/lib/python3.11/site-packages/dgl/sparse/__init__.py", line 35, in load_dgl_sparse
    raise FileNotFoundError(f"Cannot find DGL C++ sparse library at {path}")
FileNotFoundError: Cannot find DGL C++ sparse library at /Users/eslt0101/mambaforge/envs/dgl_ptm_test/lib/python3.11/site-packages/dgl/dgl_sparse/libdgl_sparse_pytorch_2.1.0.dylib
```
this is the know issue that underlay the original pinning of pytorch.   This may be a Mac vs Windows issue, if all windows systems successfully build and run.

Reply:
Interesting, which version of DGL do you have installed?